platforms/documentation/docs/src/docs/userguide/authoring-builds/tasks/lazy_configuration.adoc

```diff
@@ -321,6 +321,30 @@ $ gradle show
 include::{snippetsPath}/providers/propertyConvention/tests/propertyConvention.out[]
 ----
 
+=== Where to apply conventions from?
+
+In general, you want to apply convention to a property at configuration time, before tasks execute, so task actions are not appropriate.
+Still, there are multiple options for appropriate locations for setting a convention on a property.
+
+==== From a plugin's `apply()` method
+
+Plugin authors may configure a convention on a lazy property from a plugin's `apply()` method, while performing preliminary configuration of the property-owning task.
```

Comment:
🤔 Properties might also be part of an extension.

A common example of a task property using extension property as a fallback convention could be useful here too.

Reply:
Changed it to "while performing preliminary configuration of the owning task or extension.".